packages/gatsby-source-shopify/src/processors/product-variants.ts

```diff
@@ -1,16 +1,33 @@
-import { NodeInput } from "gatsby"
-import { pattern as idPattern } from "../node-builder"
+import { NodeInput, SourceNodesArgs } from "gatsby"
+import { createNodeId, pattern as idPattern } from "../node-builder"
 
 export function productVariantsProcessor(
   objects: BulkResults,
-  builder: NodeBuilder
+  builder: NodeBuilder,
+  gatsbyApi: SourceNodesArgs,
+  pluginOptions: ShopifyPluginOptions
 ): Array<Promise<NodeInput>> {
-  const objectsToBuild = objects.filter(obj => {
+  const filteredObjects = objects.filter(obj => {
     const [, remoteType] = obj.id.match(idPattern) || []
 
     return remoteType !== `Product`
   })
 
+  const objectsToBuild = filteredObjects.map(obj => {
+    const [, remoteType] = obj.id.match(idPattern) || []
+
+    if (remoteType === `InventoryLevel`) {
```

Comment:
I'm thinking about refactoring this so that we only need to loop through the objects one time, rather than looping through each `ProductVariant` twice here and pulling out the `remoteType`.

Reply:
@DanielSLew you could do it with reduce but I bet the performance difference is tiny

Reply:
Yep, I think it looks a lot cleaner with reduce as well. Probably not a ton of performance diff, but takes it from O(2N) to O(N).